### PR TITLE
Install refactor

### PR DESCRIPTION
**Pull request by downiec**
This is an update to the installation process which seeks to consolidate deployment into a single source. Adds a new script: scripts/deploy_update.py which when run, will automatically update/create a corresponding Dockerfile and install_script.sh which contain the appropriate updated dependencies. The Dockerfile and install script can be further updated by modifying their corresponding templates (all located in the deploy folder). This PR also includes updates to the circleci config made by Lina, which will make circleci use the latest docker image for chrome and firefox testing. It will then use the same base image needed for CDAT and will run the install script to install vcdat to match what is installed locally. JupyterLab is pinned to version 1.2.1 for compatibility reasons. Migrating to JupyterLab 2.0 will be done in a later update.

### CODE REVIEW

**Review by sterlingbaldwin — APPROVED** [review]

**Review by sterlingbaldwin — APPROVED** [review]

**Comment by sterlingbaldwin** [comment]
Go ahead and merge this once the tests are passing.

File: .circleci/OLDconfig.yml [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+      mkdir $WORKDIR
+      cd $WORKDIR
+      wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O Miniconda3-latest-Linux-x86_64.sh
+      bash ./Miniconda3-latest-Linux-x86_64.sh -b -p miniconda
+
+  - &setup_jupyter_vcdat
+    name: setup_jupyter_vcdat
+    command: |
+      source $WORKDIR/miniconda/etc/profile.d/conda.sh
+      ./install_script.sh -v -d
```
Comment (muryanto1) [inline .circleci/OLDconfig.yml]: I think we need to add "conda activate base" before running install_script.sh.

File: tests/PageObjects/Locator.py [hunk 2]
```diff
@@ -25,6 +25,23 @@ def __init__(self, driver, loc, loc_type, descr, multiple, *reqs):
...
+            name = "[{}]_not_found.png".format(self.description)
+            i = 0  # Value to add to filename to prevent overriding
+            while os.path.exists(name):
+                name = "[{}]_not_found{}.png".format(self.description, i)
+                i += 1
+        else:
+            name = "element_not_found.png"
+            while os.path.exists(name):
+                name = "element_not_found{}.png".format(i)
+                i += 1
```
Comment (muryanto1) [inline tests/PageObjects/Locator.py:42]: i is not defined in this scope.

File: tests/PageObjects/Locator.py [hunk 3]
```diff
@@ -25,6 +25,23 @@ def __init__(self, driver, loc, loc_type, descr, multiple, *reqs):
 
     """--------------- Helper Functions ---------------"""
 
+    def __save_screenshot__(self, name=""):
+        if name == "" and self.description != "":
+            name = "[{}]_not_found.png".format(self.description)
+            i = 0  # Value to add to filename to prevent overriding
+            while os.path.exists(name):
+                name = "[{}]_not_found{}.png".format(self.description, i)
```
Comment (muryanto1) [inline tests/PageObjects/Locator.py]: line 33 and 34: should we increment i first?

File: tests/PageObjects/Locator.py [hunk 4]
```diff
@@ -25,6 +25,23 @@ def __init__(self, driver, loc, loc_type, descr, multiple, *reqs):
...
+                i += 1
+        else:
+            name = "element_not_found.png"
```
Comment (muryanto1) [inline tests/PageObjects/Locator.py:36]: if the argument "name" is not empty, then we get to this else case, and you are setting 'name' to "element_not_found.png", i.e. overwriting the passed in argument. is that the intention?

**Review by muryanto1 — CHANGES_REQUESTED** [review]

File: tests/PageObjects/Locator.py (hunk 3 above)
Comment (downiec) [reply tests/PageObjects/Locator.py]: I guess we can start at 1 instead of 0.

File: tests/PageObjects/Locator.py (hunk 4 above)
Comment (downiec) [reply tests/PageObjects/Locator.py:36]: Good catch, I will fix!

**Comment by downiec** [comment]
@muryanto1 Tests are passing after I implemented your suggestions. Can you re-review and then I'll merge?

File: tests/PageObjects/Locator.py (hunk 4 above)
Comment (muryanto1) [reply tests/PageObjects/Locator.py:36]: The fix looks good. I think you can move line 31 to be at the beginning of the __save_screenshot__(),
and you can remove i = 0 from line 37 and 42.
No big deal. I can see the code should work as is. Thanks.

**Review by muryanto1 — COMMENTED** [review]
approved

**Review by muryanto1 — APPROVED** [review]

**Pull request merged**